Automattic/sensei
# Replace jQuery ui tab script, limit script loading on frontend

Fixes #2798
We were only using a jQuery UI Tabs component on the Learner profile page and a shortcode - 
bringing a lot of dependencies in for this minor usage. This was also pretty much the only script on the Sensei frontend depending on jQuery. Cleaning this up results in smaller JS bundle size and improved performance. (Other plugins like Woocommerce still load jQuery, so the speedup is not that huge for now)

The replacement code (`tabs.js`) is a bit large-ish, mostly to remain IE compatible. If we were to add more frontend scripts in the future, we can set up babel's polyfills and this file can be slimmed down a bit. 

One thing to consider if someone's customizations might depend on jQuery already being available when using the Sensei plugin. 

#### Changes proposed in this Pull Request:

* Remove `frontend/course-archive.js` script, replaced with an inline onchange handler
* Replace jQuery UI tabs with a vanilla JS script. 
* Ensure its only enqueued on the Learner Profile page and the shortcode.

#### Testing instructions:

* Make sure learner profiles are enabled
* Go to /learner/username 
* *Active Courses, Completed Courses* tabs should work
* `tabs.js` is not loaded on any other page

* Add [usercourses] shortcode to a post
* `tabs.js` is loaded and tabs work

* Open Course archive page
* *Sort by...* dropdown updates the page on selecting an item


#### Proposed changelog entry for your changes:

* Sensei no longer loads jQuery on the frontend
